tests/test_batch_prefill.py

```diff
@@ -0,0 +1,48 @@
+import torch
+import pytest
+from flashinfer import BatchPrefillWithPagedKVCacheWrapper
+
+
+@pytest.mark.parametrize("dtype", [torch.float16, torch.bfloat16])
+def test_kv_scale_forwarding_effect(dtype):
+    torch.manual_seed(42)
+
+    B, H_QO, H_KV, N_CTX, HEAD_DIM, PAGE_SIZE = 1, 1, 1, 8, 64, 16
+    max_num_pages = (N_CTX + PAGE_SIZE - 1) // PAGE_SIZE
+
+    # Create paged KV cache
+    k_cache = torch.randn(max_num_pages, PAGE_SIZE, H_KV, HEAD_DIM, dtype=dtype, device="cuda")
+    v_cache = torch.randn(max_num_pages, PAGE_SIZE, H_KV, HEAD_DIM, dtype=dtype, device="cuda")
+    paged_kv_cache = (k_cache, v_cache)
+
+    # Create query tensor and indptrs
+    q = torch.randn(N_CTX, H_QO, HEAD_DIM, dtype=dtype, device="cuda")
+    qo_indptr = torch.tensor([0, N_CTX], dtype=torch.int32, device="cuda")
+    paged_kv_indptr = torch.tensor([0, max_num_pages], dtype=torch.int32, device="cuda")
+    paged_kv_indices = torch.arange(max_num_pages, dtype=torch.int32, device="cuda")
+    paged_kv_last_page_len = torch.tensor([N_CTX % PAGE_SIZE or PAGE_SIZE], dtype=torch.int32, device="cuda")
+
+    workspace_buffer = torch.empty(16 * 1024 * 1024, dtype=torch.uint8, device="cuda")
+    wrapper = BatchPrefillWithPagedKVCacheWrapper(workspace_buffer)
+
+    wrapper.plan(
+        qo_indptr,
+        paged_kv_indptr,
+        paged_kv_indices,
+        paged_kv_last_page_len,
+        H_QO,
+        H_KV,
+        HEAD_DIM,
+        PAGE_SIZE,
+        causal=True,
+        q_data_type=dtype,
+        kv_data_type=dtype,
+    )
+
+    out1, _ = wrapper.forward_return_lse(q, paged_kv_cache, k_scale=0.1, v_scale=0.1)
+    out2, _ = wrapper.forward_return_lse(q, paged_kv_cache, k_scale=2.0, v_scale=2.0)
+
+    assert not torch.allclose(out1, out2, atol=1e-3), (
```

Comment:
this test still looks week to me, can we add another set of tasks like:
1. wrapper.forward_return_lse(q, paged_kv_cache, k_scale=k_scale) should be same as wrapper.forward_return_lse(q * k_scale, paged_kv_cache)
2. wrapper.forward_return_lse(q, paged_kv_cache, v_scale=v_scale) should be same as wrapper.forward_return_lse(q, paged_kv_cache) * v_scale
3. wrapper.forward_return_lse(q, paged_kv_cache, k_scale=k_scale, v_scale=v_scale) should be same as wrapper.forward_return_lse(q * k_scale, paged_kv_cache) * v_scale

and varies the value of k_scale and v_scale